File: Fundamentals.Of.Python.Data.Structures/1st_code/Ch_3_Student_Files/Case Study/test_bubble_sort.py
#!/usr/bin/env python3
"""
Testing the Sorting algorithms
Roman Yasinovskyy, 2017
Karina E. Hoff, 2018
"""


# Specifies the absolute path to the pythonds3 module
import os
import sys

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))

import pytest
from random import randint
from bubbleSort import bubbleSort
from bubbleSortWithTweak import bubbleSortWithTweak


class TestSortingMethods:
    """Testing the sorting algorithms"""

    @pytest.fixture(autouse=True)
    def setup_class(self):
        """Setting up"""
        self.lst_to_sort = [randint(100, 999) for _ in range(100)]
        self.test_lst = self.lst_to_sort[:]
        self.test_lst.sort()

    def test_bubble_sort(self):
        """Testing BubbleSort"""
        bubbleSort(self.lst_to_sort)
        assert self.lst_to_sort == self.test_lst

    def test_bubble_sort_with_Tweak(self):
        """Testing BubbleSortWithTweak"""
        bubbleSortWithTweak(self.lst_to_sort)
        assert self.lst_to_sort == self.test_lst


if __name__ == "__main__":
    pytest.main(["test_bubble_sort.py"])

File: Fundamentals.Of.Python.Data.Structures/cxx/1st_code/Ch_3_Student_Files/Case Study/bubble_sort/test_bubble_sort.cpp
#include <gtest/gtest.h>
#include <iostream>
#include <algorithm>
#include "random.hpp"
#include "bubble_sort.hpp"
#include "bubble_sort_with_tweak.hpp"
#include "sort_helper.hpp"

class TestSortingMethods: public testing::Test {
protected:
    virtual void SetUp()
    {
        for (int i=0; i < 100; i++) {
            lst_to_sort.push_back(randint(100, 999));
        }
        test_lst = lst_to_sort;
        std::sort(test_lst.begin(), test_lst.end());
    }

    virtual void TearDown()
    {
        print(test_lst);
    }
    
    std::vector<int> lst_to_sort;
    std::vector<int> test_lst;
};

TEST_F(TestSortingMethods, test_bubble_sort) {
    bubbleSort(lst_to_sort);
    EXPECT_EQ(lst_to_sort, test_lst);
}

TEST_F(TestSortingMethods, test_bubble_sort_with_tweak) {
    bubbleSortWithTweak(lst_to_sort);
    EXPECT_EQ(lst_to_sort, test_lst);
}

int main(int argc, char* argv[])
{
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}
